src/Sharpbrake.Client/NoticeBuilder.cs

```diff
@@ -154,6 +154,33 @@ public static string ToJsonString(Notice notice)
         }
 
         /// <summary>
+        /// Gets JSON string for the truncated instance of <see cref="Notice"/>.
+        /// </summary>
+        public static string ToJsonStringWithTruncation(Notice notice)
+        {
+            const int noticeLengthMax = 64000;
+            const int stringLengthMax = 1024;
+
+            var json = ToJsonString(notice);
+            var level = 0;
+
+            while (json.Length > noticeLengthMax && level < 8)
+            {
+                // each level reduces the string limit by a half
+                var stringLimit = stringLengthMax / (int) Math.Pow(2, level);
+
+                notice.EnvironmentVars = Utils.TruncateParameters(notice.EnvironmentVars, stringLimit);
+                notice.Params = Utils.TruncateParameters(notice.Params, stringLimit);
+                notice.Session = Utils.TruncateParameters(notice.Session, stringLimit);
+
+                json = ToJsonString(notice);
```

Comment:
Mmm... there were a couple of reasons for that. However, none of them is strong enough per se. First of all, it could be considered as breaking changes, also some unit tests use it as a helper method and for me, it looked like some kind of imbalance between these two methods. But now I think that we can just put the logic of truncation into the regular `ToJsonString` method. In the method `SetErrorEntries` we limit to 3 inner exceptions but we didn't call it `SetErrorEntriesWithExceptionTruncation` then why we would like to differentiate `ToJsonString` with and w\o truncation, right?